src/targets/gpu/mlir.cpp

```diff
@@ -498,11 +503,24 @@ struct mlir_program
         return ins->get_shape();
     }
 
+    void set_sym_name(const module& m)
```

Comment:
I think this should just return the symbol name:

```cpp
static std::string get_symbol_name(const module& m)
    {
        for(auto ins : iterator_for(m))
        {
            if(ins->name() == "convolution" or ins->name() == "dot")
            {
                return "mlir_" + ins->name();
            }
        }
        return "main";
    }
```

Reply:
And then do `sym_nam = get_symbol_name(m)` in `parse` method.

Reply:
done